x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/60_semantic_text_inference_update.yml

```diff
@@ -440,6 +440,89 @@ setup:
+  - requires:
+      cluster_features: semantic_text.partial_update_in_object_fix
+      reason: Partial update when semantic text field is in an object fixed in 8.15.5, 8.16.2, and 8.17.0
+
+  - do:
+      indices.create:
+        index: test-in-object-index
+        body:
+          mappings:
+            properties:
+              object.sparse_field:
+                type: semantic_text
+                inference_id: sparse-inference-id
+              object.sparse_source_field:
+                type: text
+                copy_to: object.sparse_field
+              object.dense_field:
+                type: semantic_text
+                inference_id: dense-inference-id
+              object.dense_source_field:
+                type: text
+                copy_to: object.dense_field
+
+  - do:
+      index:
+        index: test-in-object-index
+        id: doc_1
+        body:
+          object:
+            sparse_field: "sparse data 1"
+            sparse_source_field: "sparse data 2"
+            dense_field: "dense data 1"
+            dense_source_field: "dense data 2"
+
+  - do:
+      get:
+        index: test-in-object-index
+        id: doc_1
+
+  - match: { _source.object.sparse_field.text: "sparse data 1" }
+  - length: { _source.object.sparse_field.inference.chunks: 2 }
+  - match: { _source.object.sparse_field.inference.chunks.1.text: "sparse data 2" }
+  - exists: _source.object.sparse_field.inference.chunks.1.embeddings
+  - match: { _source.object.dense_field.text: "dense data 1" }
+  - length: { _source.object.dense_field.inference.chunks: 2 }
+  - match: { _source.object.dense_field.inference.chunks.1.text: "dense data 2" }
+  - exists: _source.object.dense_field.inference.chunks.1.embeddings
+
+  - do:
+      update:
+        index: test-in-object-index
+        id: doc_1
+        body:
+          doc: { "object.sparse_source_field": "sparse data 3", "object.dense_source_field": "dense data 3" }
+
+  - do:
+      get:
+        index: test-in-object-index
+        id: doc_1
+
+  - match: { _source.object.sparse_field.text: "sparse data 1" }
+  - length: { _source.object.sparse_field.inference.chunks: 3 }
+  - match: { _source.object.sparse_field.inference.chunks.1.text: "sparse data 2" }
+  - exists: _source.object.sparse_field.inference.chunks.1.embeddings
+  - match: { _source.object.sparse_field.inference.chunks.2.text: "sparse data 3" }
+  - exists: _source.object.sparse_field.inference.chunks.2.embeddings
+  - match: { _source.object.dense_field.text: "dense data 1" }
+  - length: { _source.object.dense_field.inference.chunks: 3 }
+  - match: { _source.object.dense_field.inference.chunks.1.text: "dense data 2" }
+  - exists: _source.object.dense_field.inference.chunks.1.embeddings
+  - match: { _source.object.dense_field.inference.chunks.2.text: "dense data 3" }
+  - exists: _source.object.dense_field.inference.chunks.2.embeddings
```

Comment:
Notice how the updated values for `object.sparse_source_field` and `object.dense_source_field` are appended instead of overwritten. This is because they are defined using a flat path instead of a nested path, resulting in a merged `_source` with both values:

```json
{
  "object": {
    "sparse_source_field": "sparse data 2",
    "dense_source_field": "dense data 2",
  },
  "object.sparse_source_field": "sparse data 3",
  "object.dense_source_field": "dense data 3"
}
```